src/Swarm/Util.hs

```diff
@@ -210,6 +215,30 @@ readFileMayT = catchIO . T.readFile
 catchIO :: IO a -> IO (Maybe a)
 catchIO act = (Just <$> act) `catchIOError` (\_ -> return Nothing)
 
+getDataDirSafe :: IO (Maybe FilePath)
+getDataDirSafe = do
+  d <- getDataDir
+  de <- doesDirectoryExist d
+  if de
+    then return $ Just d
+    else do
+      xd <- (</> "data") <$> getSwarmDataPath False
+      xde <- doesDirectoryExist xd
+      return $ if xde then Just xd else Nothing
+
+getDataFileNameSafe :: FilePath -> IO (Maybe FilePath)
+getDataFileNameSafe name = do
+  dir <- getDataDirSafe
+  case dir of
+    Nothing -> return Nothing
+    Just d -> do
+      let fp = d </> name
+      fe <- doesFileExist fp
+      return $ if fe then Just fp else Nothing
+
+dataFileNotFound :: Text
+dataFileNotFound = T.pack ""
```

Comment:
Note to self: Add some meaningful error message teling the user that the file/directory was not found and that they might want to download the data archive and extract it to `<xdg data>/data`.